Description: Adapt to PHPUnit 8.x and 9.x API.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/IdentityTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/IdentityTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/IdentityTest.php
@@ -23,13 +23,13 @@
  * @link      http://pear.horde.org/index.php?package=Prefs
  * @package   Prefs
  */
-class Horde_Prefs_Unit_IdentityTest extends PHPUnit_Framework_TestCase
+class Horde_Prefs_Unit_IdentityTest extends Horde_Test_Case
 {
     private $identity;
 
     /**
      */
-    public function setUp()
+    public function setUp(): void
     {
         $this->identity = new Horde_Prefs_Identity(array(
             'prefs' => new Horde_Prefs(
@@ -55,7 +55,7 @@ class Horde_Prefs_Unit_IdentityTest exte
     {
         $this->identity->add(array());
 
-        $this->assertInternalType('array', $this->identity->get(0));
+        $this->assertIsArray($this->identity->get(0));
         $this->assertNull($this->identity->get(1));
     }
 
@@ -89,7 +89,7 @@ class Horde_Prefs_Unit_IdentityTest exte
     {
         $this->identity->add(array());
 
-        $this->assertInternalType('array', $this->identity[0]);
+        $this->assertIsArray($this->identity[0]);
         $this->assertNull($this->identity[1]);
     }
 
@@ -99,7 +99,7 @@ class Horde_Prefs_Unit_IdentityTest exte
     {
         $this->identity->add(array());
 
-        $this->assertInternalType('array', $this->identity[0]);
+        $this->assertIsArray($this->identity[0]);
 
         unset($this->identity[0]);
 
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/FileTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/FileTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/FileTest.php
@@ -25,26 +25,23 @@
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  * @link     http://pear.horde.org/index.php?package=Prefs
  */
-class Horde_Prefs_Unit_Storage_FileTest extends PHPUnit_Framework_TestCase
+class Horde_Prefs_Unit_Storage_FileTest extends Horde_Test_Case
 {
-    /**
-     * @expectedException InvalidArgumentException
-     */
     public function testMissingDirectory()
     {
+        $this->expectException('InvalidArgumentException');
         $b = new Horde_Prefs_Storage_File('nobody');
     }
 
-    /**
-     * @expectedException InvalidArgumentException
-     */
     public function testInvalidDirectory()
     {
+        $this->expectException('InvalidArgumentException');
         $b = new Horde_Prefs_Storage_File('nobody', array('directory' => __DIR__ . '/DOES_NOT_EXIST'));
     }
 
     public function testConstruction()
     {
+        $this->expectNotToPerformAssertions();
         $b = new Horde_Prefs_Storage_File('nobody', array('directory' => Horde_Util::createTempDir()));
     }
 }
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/KolabImapLogTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/KolabImapLogTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/KolabImapLogTest.php
@@ -27,7 +27,7 @@
  */
 class Horde_Prefs_Unit_Storage_KolabImapLogTest extends Horde_Test_Log
 {
-    public function setUp()
+    public function setUp(): void
     {
         if (!class_exists('Horde_Kolab_Storage_Factory')) {
             $this->markTestSkipped('Horde_Kolab_Storage package is missing');
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/KolabImapTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/KolabImapTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/KolabImapTest.php
@@ -21,9 +21,9 @@
  * @license  http://www.horde.org/licenses/lgpl21 LGPL 2.1
  * @link     http://pear.horde.org/index.php?package=Prefs
  */
-class Horde_Prefs_Unit_Storage_KolabImapTest extends PHPUnit_Framework_TestCase
+class Horde_Prefs_Unit_Storage_KolabImapTest extends Horde_Test_Case
 {
-    public function setUp()
+    public function setUp(): void
     {
         if (!class_exists('Horde_Kolab_Storage_Factory')) {
             $this->markTestSkipped('Horde_Kolab_Storage package is missing');
@@ -31,16 +31,15 @@ class Horde_Prefs_Unit_Storage_KolabImap
         $_SESSION = array();
     }
 
-    /**
-     * @expectedException InvalidArgumentException
-     */
     public function testMissingStorage()
     {
+        $this->expectException('InvalidArgumentException');
         $b = new Horde_Prefs_Storage_KolabImap('nobody');
     }
 
     public function testConstruction()
     {
+        $this->expectNotToPerformAssertions();
         $b = new Horde_Prefs_Storage_KolabImap(
             'nobody', array('kolab' => $this->_createDefaultStorage())
         );
@@ -163,7 +162,7 @@ class Horde_Prefs_Unit_Storage_KolabImap
                     'data' => array('queryset' => 'horde'),
                 ),
                 'cache'  => new Horde_Cache(new Horde_Cache_Storage_Mock()),
-                'logger' => $this->getMock('Horde_Log_Logger'),
+                'logger' => $this->getMockBuilder('Horde_Log_Logger')->getMock(),
             )
         );
         return $factory->create();
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Base.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Base.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Base.php
@@ -55,7 +55,7 @@ class Horde_Prefs_Test_Sql_Base extends
         );
     }
 
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         $logger = new Horde_Log_Logger(new Horde_Log_Handler_Cli());
         //self::$db->setLogger($logger);
@@ -81,7 +81,7 @@ class Horde_Prefs_Test_Sql_Base extends
         self::$prefs = new Horde_Prefs_Storage_Sql('joe', array('db' => self::$db));
     }
 
-    public static function tearDownAfterClass()
+    public static function tearDownAfterClass(): void
     {
         self::$prefs = null;
         if (self::$db) {
@@ -97,7 +97,7 @@ class Horde_Prefs_Test_Sql_Base extends
         }
     }
 
-    public function setUp()
+    public function setUp(): void
     {
         if (!self::$db) {
             $this->markTestSkipped(self::$reason);
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/MysqlTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/MysqlTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/MysqlTest.php
@@ -15,7 +15,7 @@ require_once __DIR__ . '/Base.php';
  */
 class Horde_Prefs_Unit_Storage_Sql_MysqlTest extends Horde_Prefs_Test_Sql_Base
 {
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         if (!extension_loaded('mysql')) {
             self::$reason = 'No mysql extension';
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/MysqliTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/MysqliTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/MysqliTest.php
@@ -15,7 +15,7 @@ require_once __DIR__ . '/Base.php';
  */
 class Horde_Prefs_Unit_Storage_Sql_MysqliTest extends Horde_Prefs_Test_Sql_Base
 {
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         if (!extension_loaded('mysqli')) {
             self::$reason = 'No mysqli extension';
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Oci8Test.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Oci8Test.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Oci8Test.php
@@ -15,7 +15,7 @@ require_once __DIR__ . '/Base.php';
  */
 class Horde_Prefs_Unit_Storage_Sql_Oci8Test extends Horde_Prefs_Test_Sql_Base
 {
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         if (!extension_loaded('oci8')) {
             self::$reason = 'No oci8 extension';
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/MysqlTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/MysqlTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/MysqlTest.php
@@ -15,7 +15,7 @@ require_once __DIR__ . '/../Base.php';
  */
 class Horde_Prefs_Unit_Storage_Sql_Pdo_MysqlTest extends Horde_Prefs_Test_Sql_Base
 {
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         if (!extension_loaded('pdo') ||
             !in_array('mysql', PDO::getAvailableDrivers())) {
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/PgsqlTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/PgsqlTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/PgsqlTest.php
@@ -15,7 +15,7 @@ require_once __DIR__ . '/../Base.php';
  */
 class Horde_Prefs_Unit_Storage_Sql_Pdo_PgsqlTest extends Horde_Prefs_Test_Sql_Base
 {
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         if (!extension_loaded('pdo') ||
             !in_array('pgsql', PDO::getAvailableDrivers())) {
Index: php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/SqliteTest.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/SqliteTest.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/test/Horde/Prefs/Unit/Storage/Sql/Pdo/SqliteTest.php
@@ -15,7 +15,7 @@ require_once __DIR__ . '/../Base.php';
  */
 class Horde_Prefs_Unit_Storage_Sql_Pdo_SqliteTest extends Horde_Prefs_Test_Sql_Base
 {
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
         $factory_db = new Horde_Test_Factory_Db();
 
