From: Anton Gladky <gladk@debian.org>
Date: Thu, 9 Nov 2023 16:30:53 +0200
Subject: Fix autopkgtest failure due to php8.1 changes

Last-Update: 2022-12-16
---
 Horde_Prefs-2.9.0/lib/Horde/Prefs.php | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Horde_Prefs-2.9.0/lib/Horde/Prefs.php b/Horde_Prefs-2.9.0/lib/Horde/Prefs.php
index 5d472b4..891dfc8 100644
--- a/Horde_Prefs-2.9.0/lib/Horde/Prefs.php
+++ b/Horde_Prefs-2.9.0/lib/Horde/Prefs.php
@@ -527,21 +527,25 @@ class Horde_Prefs implements ArrayAccess
 
     /* ArrayAccess methods. */
 
+    #[\ReturnTypeWillChange]
     public function offsetExists($offset)
     {
         return !is_null($this->getValue($offset));
     }
 
+    #[\ReturnTypeWillChange]
     public function offsetGet($offset)
     {
         return $this->getValue($offset);
     }
 
+    #[\ReturnTypeWillChange]
     public function offsetSet($offset, $value)
     {
         $this->setValue($offset, $value);
     }
 
+    #[\ReturnTypeWillChange]
     public function offsetUnset($offset)
     {
         $this->remove($offset);
