Index: php-horde-prefs/Horde_Prefs-2.9.0/lib/Horde/Prefs/Identity.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/lib/Horde/Prefs/Identity.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/lib/Horde/Prefs/Identity.php
@@ -98,11 +98,13 @@ implements ArrayAccess, Countable, Itera
         $this->_prefs = $params['prefs'];
         $this->_user = $params['user'];
 
-        if (!($this->_identities = @unserialize($this->_prefs->getValue($this->_prefnames['identities'])))) {
-            $this->_identities = $this->_prefs->getDefault($this->_prefnames['identities']);
-        }
+        foreach ($this->_prefs as $val) {
+            if (!($this->_identities = @unserialize($val->getValue($this->_prefnames['identities'])))) {
+                $this->_identities = $val->getDefault($this->_prefnames['identities']);
+            }
 
-        $this->setDefault($this->_prefs->getValue($this->_prefnames['default_identity']));
+            $this->setDefault($val->getValue($this->_prefnames['default_identity']));
+        }
     }
 
     /**
@@ -415,7 +417,7 @@ implements ArrayAccess, Countable, Itera
     /**
      * @since 2.7.0
      */
-    public function offsetExists($offset)
+    public function offsetExists($offset): bool
     {
         return isset($this->_identities[$offset]);
     }
@@ -423,7 +425,7 @@ implements ArrayAccess, Countable, Itera
     /**
      * @since 2.7.0
      */
-    public function offsetGet($offset)
+    public function offsetGet($offset): mixed
     {
         return $this->get($offset);
     }
@@ -431,7 +433,7 @@ implements ArrayAccess, Countable, Itera
     /**
      * @since 2.7.0
      */
-    public function offsetSet($offset, $value)
+    public function offsetSet($offset, $value): void
     {
         // $value is ignored.
         $this->set($offset);
@@ -440,7 +442,7 @@ implements ArrayAccess, Countable, Itera
     /**
      * @since 2.7.0
      */
-    public function offsetUnset($offset)
+    public function offsetUnset($offset): void
     {
         $this->delete($offset);
     }
@@ -450,7 +452,7 @@ implements ArrayAccess, Countable, Itera
     /**
      * @since 2.7.0
      */
-    public function count()
+    public function count(): int
     {
         return count($this->_identities);
     }
@@ -460,7 +462,7 @@ implements ArrayAccess, Countable, Itera
     /**
      * @since 2.7.0
      */
-    public function getIterator()
+    public function getIterator(): Traversable
     {
         return new ArrayIterator($this->_identities);
     }
Index: php-horde-prefs/Horde_Prefs-2.9.0/lib/Horde/Prefs/Scope.php
===================================================================
--- php-horde-prefs.orig/Horde_Prefs-2.9.0/lib/Horde/Prefs/Scope.php
+++ php-horde-prefs/Horde_Prefs-2.9.0/lib/Horde/Prefs/Scope.php
@@ -276,6 +276,7 @@ class Horde_Prefs_Scope implements Itera
 
     /**
      */
+    #[\ReturnTypeWillChange]
     public function current()
     {
         return $this->_fromInternal($this->key());
@@ -283,6 +284,7 @@ class Horde_Prefs_Scope implements Itera
 
     /**
      */
+    #[\ReturnTypeWillChange]
     public function key()
     {
         return key($this->_prefs);
@@ -290,6 +292,7 @@ class Horde_Prefs_Scope implements Itera
 
     /**
      */
+    #[\ReturnTypeWillChange]
     public function next()
     {
         return next($this->_prefs);
@@ -297,6 +300,7 @@ class Horde_Prefs_Scope implements Itera
 
     /**
      */
+    #[\ReturnTypeWillChange]
     public function rewind()
     {
         return reset($this->_prefs);
@@ -304,6 +308,7 @@ class Horde_Prefs_Scope implements Itera
 
     /**
      */
+    #[\ReturnTypeWillChange]
     public function valid()
     {
         return !is_null(key($this->_prefs));
